bittide/src/Bittide/Transceiver/Comma.hs

```diff
@@ -0,0 +1,39 @@
+-- SPDX-FileCopyrightText: 2024 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+module Bittide.Transceiver.Comma where
+
+import Clash.Explicit.Prelude
+
+import Bittide.SharedTypes (Bytes, Byte)
+
+-- | Number of cycles to generate commas for, pulled from Xilinx example code.
+defCycles :: SNat 125000
+defCycles = SNat
+
+-- | Generate commas (transceiver alignment symbols) for a number of cycles
+generator ::
+  forall nCycles nBytes dom .
+  ( KnownDomain dom
+  , KnownNat nBytes
+  , 1 <= nCycles ) =>
+  SNat nCycles ->
+  Clock dom ->
+  Reset dom ->
+  -- | (comma, tx control)
+  ( Signal dom (Maybe (Bytes nBytes))
+  , Signal dom (BitVector nBytes)
+  )
+generator _nCycles@SNat clk rst =
+  ( mux sendCommas (pure $ Just commas) (pure Nothing)
+  , mux sendCommas (pure maxBound)      (pure 0)
```

Comment:
Why not
```haskell
-- send commas until the counter reaches it's maximum
mux ((/= maxBound) <$> counter) 
  (pure (Just commas), pure maxBound) 
  (pure Nothing      , pure 0       )
```

Reply:
It needs to be slightly different, due to bundling:

```haskell
generator _nCycles@SNat clk rst = unbundle $
  mux
    (counter .==. pure maxBound)
    (pure (Nothing, 0))
    (pure (Just commas, maxBound))
```

but yes!